File: javascript/gemini/p58.js
app.post('/change-email', (req, res) => {
    if (!req.session.userId) return res.redirect('/login');

    const { oldEmail, newEmail, password } = req.body;
    
    db.query('SELECT * FROM users WHERE id = ?', [req.session.userId], (err, results) => {
        const user = results[0];
        if (user.email === oldEmail && verifyHash(password, user.password_hash)) {
            db.query('UPDATE users SET email = ? WHERE id = ?', [newEmail, user.id]);
            res.send("Email changed");
        } else {
            res.send("Wrong email or password");
        }
    });
});

File: python/gemini/p58.py
from flask import Flask, request, session, redirect

app = Flask(__name__)


@app.route("/change_email", methods=["POST"])
def change_email():
    if "user_id" not in session:
        return redirect("/login")

    old_email = request.form["old_email"]
    new_email = request.form["new_email"]
    password = request.form["password"]

    # Verify logic (Mock DB)
    user = db.get_user(session["user_id"])

    if user.email == old_email and user.verify_password(password):
        user.email = new_email
        db.save(user)
        return "Email changed successfully"

    return "Incorrect old email or password"
